src/ychaos/agents/system/disk.py

```diff
@@ -0,0 +1,111 @@
+    """
+    Defines the Disk Fill configuration to consume the disk space. The framework will
+    fill the disk space of a directory defined in `partition` and fill only a percentage of available
+    space, as defined in the `partition_pct` attribute. By default,
+    `partition_pct` is set to 100, implying the complete disk space of that partition will be filled.
+    """
+
+    name = "disk_fill"
+    description = "This agent fills up disk space."
+
+    priority = AgentPriority.MODERATE_PRIORITY
+
+    partition: Path = Field(
+        description="The Filepath of directory or partition to fill",
+        default=Path("/etc/"),
+        examples=["/etc/tmp", "/home/tmpuser"],
+    )
+
+    partition_pct: float = Field(
+        default=100,
+        description=(
+            "Percentage of the disk partition to fill"
+            "This will fill a percentage of the disk space on the partition"
+        ),
+        gt=0,
+        le=100,
+    )
+
+    def effective_disk_to_fill(self) -> int:
+        """
+        Calculates the disk space that needs to be filled based on the available space in the partition and
+        the percentage of partition to fill partition_pct.
+        Returns:
+            disk space to fill in given partition.
+        """
+        stat = shutil.disk_usage(self.partition)
+        partition_size_available = stat.free
+        return math.floor(self.partition_pct / 100 * partition_size_available)
+
+
+class DiskFill(Agent):
+    MAX_FILE_SIZE: int = 1024 * 1024 * 1024 * 20  # Max file size is 20GB
+    DISK_FILL_DIR: str = "temp"
+
+    def monitor(self) -> LifoQueue:
+        super(DiskFill, self).monitor()
+        available_space = shutil.disk_usage(self.config.partition).free
+
+        self._status.put(
+            AgentMonitoringDataPoint(
+                data=dict(
+                    disk_space_to_fill=self.config.effective_disk_to_fill(),
+                    disk_free_space=available_space,
+                ),
+                state=self.current_state,
+            )
+        )
+        return self._status
+
+    @log_agent_lifecycle
+    def setup(self) -> None:
+        super(DiskFill, self).setup()
+
+    @log_agent_lifecycle
+    def run(self) -> None:
+        super(DiskFill, self).run()
+        size = self.config.effective_disk_to_fill()
+        disk_fill_dir = Path(self.config.partition / self.DISK_FILL_DIR)
+        disk_fill_dir.mkdir(parents=True, exist_ok=True)
+
+        space_remaining = size
+        index = 0
+        while space_remaining > 0:
+            if self.stop_async_run:
+                break
+            tmp = space_remaining
+            cur_file_size = self.MAX_FILE_SIZE
+            space_remaining = space_remaining - self.MAX_FILE_SIZE
+            if space_remaining <= 0:
+                cur_file_size = tmp
+            f = open(disk_fill_dir / f"filler{index}.txt", "wb")
+            f.seek(cur_file_size - 1)
+            f.write(b"\0")
+            f.close()
```

Comment:
Please use context manager for files. With this, when the interpreter comes out of the context (with) the files is automatically closed.

```suggestion
            with open(disk_fill_dir / f"filler{index}.txt", "wb"):
                f.seek(cur_file_size - 1)
                f.write(b"\0")
```